src/test/java/org/folio/rest/api/RequestBatchAPITest.java

```diff
@@ -0,0 +1,314 @@
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import org.folio.rest.jaxrs.model.Request;
+import org.folio.rest.support.ApiTests;
+import org.folio.rest.support.JsonResponse;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+import org.folio.rest.support.TextResponse;
+import org.folio.rest.support.builders.RequestRequestBuilder;
+import org.hamcrest.Matcher;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class RequestBatchAPITest extends ApiTests {
+
+  @Before
+  public void beforeEach() throws Exception {
+    StorageTestSuite.deleteAll(requestStorageUrl());
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs("request");
+  }
+
+  @Test
+  public void canUpdateRequestPositionsInBatch() throws Exception {
+    UUID itemId = UUID.randomUUID();
+
+    JsonObject firstRequest = createRequestForItemAtPosition(itemId, 1);
+    JsonObject secondRequest = createRequestForItemAtPosition(itemId, 2);
+
+    ReorderRequest firstReorderRequest = new ReorderRequest(firstRequest, 2);
+    ReorderRequest secondReorderRequest = new ReorderRequest(secondRequest, 1);
+
+    reorderRequests(
+      firstReorderRequest,
+      secondReorderRequest
+    );
+
+    assertRequestsUpdated(itemId, firstReorderRequest, secondReorderRequest);
+  }
+
+  @Test
+  public void canCloseRequestsInBatch() throws Exception {
+    UUID itemId = UUID.randomUUID();
+
+    JsonObject firstRequest = createRequestForItemAtPosition(itemId, 1);
+    JsonObject secondRequest = createRequestForItemAtPosition(itemId, 2);
+
+    firstRequest.put("status", Request.Status.CLOSED_FILLED.value());
+    firstRequest.remove("position");
+    secondRequest.put("status", Request.Status.CLOSED_FILLED.value());
+    secondRequest.remove("position");
+
+    CompletableFuture<Response> postCompleted = new CompletableFuture<>();
+    client.post(batchRequestStorageUrl(),
+      buildBatchUpdateRequest(firstRequest, secondRequest),
+      TENANT_ID,
+      ResponseHandler.empty(postCompleted)
+    );
+
+    Response response = postCompleted.get(5, TimeUnit.SECONDS);
+    assertThat(response.getStatusCode(), is(201));
+
+    JsonObject allRequestsForItem = getAllRequestsForItem(itemId);
+    assertThat(allRequestsForItem.getInteger("totalRecords"), is(2));
+
+    JsonArray allRequests = allRequestsForItem.getJsonArray("requests");
+    assertThat(firstRequest, is(allRequests.getJsonObject(0)));
+    assertThat(secondRequest, is(allRequests.getJsonObject(1)));
+  }
+
+  @Test
+  public void canUpdateRequestFulfilmentPreferenceInBatch() throws Exception {
```

Comment:
Why would the fulfilment preferences need to be updated in a batch?

Reply:
to make sure that we still able to update different from `position` property without issues.

Reply:
Ok. thanks for explaining that to me